src/applications/personalization/profile-2/components/ProfileWrapper.jsx

```diff
@@ -1,28 +1,93 @@
 import { connect } from 'react-redux';
+import LoadingIndicator from '@department-of-veterans-affairs/formation-react/LoadingIndicator';
 
 import RequiredLoginView from 'platform/user/authorization/components/RequiredLoginView';
 import backendServices from 'platform/user/profile/constants/backendServices';
 
+import {
+  fetchMilitaryInformation as fetchMilitaryInformationAction,
+  fetchHero as fetchHeroAction,
+} from 'applications/personalization/profile360/actions';
+
 import ProfileHeader from './ProfileHeader';
 import ProfileSideNav from './ProfileSideNav';
 
-const ProfileWrapper = ({ children, user }) => (
-  <RequiredLoginView serviceRequired={backendServices.USER_PROFILE} user={user}>
-    <ProfileHeader />
-    <div className="usa-grid usa-grid-full">
-      <div className="usa-width-one-fourth">
-        <ProfileSideNav />
-      </div>
-      <div className="usa-width-three-fourths">{children}</div>
+const ProfileWrapper = ({
+  children,
+  fetchFullName,
+  fetchMilitaryInformation,
+  showLoader,
+  user,
+}) => {
+  useEffect(
+    () => {
+      fetchMilitaryInformation();
+      fetchFullName();
+    },
+    [fetchMilitaryInformation, fetchFullName],
+  );
+
+  // content to show if the component is waiting for data to load
+  const loadingContent = (
+    <div className="vads-u-margin-y--5">
+      <LoadingIndicator setFocus message="Loading your information..." />
     </div>
-  </RequiredLoginView>
-);
+  );
+
+  // content to show after data has loaded
+  // note that `children` will be passed in via React Router.
+  const mainContent = (
+    <>
+      <ProfileHeader />
+      <div className="usa-grid usa-grid-full">
+        <div className="usa-width-one-fourth">
+          <ProfileSideNav />
+        </div>
+        <div className="usa-width-three-fourths">{children}</div>
+      </div>
+    </>
+  );
+
+  const renderContent = () => {
+    if (showLoader) {
+      return loadingContent;
+    }
+    return mainContent;
+  };
+
+  return (
+    <RequiredLoginView
+      serviceRequired={backendServices.USER_PROFILE}
+      user={user}
+    >
+      {renderContent()}
+    </RequiredLoginView>
+  );
+};
+
+const mapStateToProps = state => {
+  // this piece of state will be set if the call to load military info succeeds or fails
+  const hasLoadedMilitaryInformation =
+    state.vaProfile?.militaryInformation?.serviceHistory;
+  // this piece of state will be set if the call to load name info succeeds or fails
+  const hasLoadedFullName = state.vaProfile?.hero?.userFullName;
+  const isLoading = !hasLoadedMilitaryInformation || !hasLoadedFullName;
+
+  return {
+    user: state.user,
+    showLoader: isLoading,
+  };
+};
 
-const mapStateToProps = state => ({
-  user: state.user,
-});
+const mapDispatchToProps = {
+  fetchMilitaryInformation: fetchMilitaryInformationAction,
+  fetchFullName: fetchHeroAction,
+};
 
-export { ProfileWrapper };
+export { ProfileWrapper, mapStateToProps };
```

Comment:
is this double export necessary with ln90 exporting ProfileWrapper as default?

Reply:
I do this so that I can test the `ProfileWrapper` directly without mocking the store, which is something that we do with most of our Redux-connected components.